scripts/heratape_backup.py

```diff
@@ -0,0 +1,330 @@
+            sys.exit()
+        #if theres more than one, choose the topmost empty slot
+        if len(empty_slots)==1:
+            emptyslot = empty_slots
+        else:
+            emptyslot = empty_slots[0]
+        #unload working drive to this empty slot
+        logger.info(f'unloading drive {mydrive} back to slot {empyslot}') 
+        unload_tape(mydrive,emptyslot)
+        logger.info('tape unloaded')
+        #select a fresh tape
+        #  this might require human intervention, which we pessimistically assume might be the case until 
+        #  proven otherwise
+        waitingforhuman = True
+        while(waitingforhuman):
+            update_tapes(testing=TESTING) #check for any new tapes in the library
+            newslot,newtape_id,newtape_usage = select_empty_tape(mydrive,testing=TESTING)
+            if newslot is None:
+                if mydrive==0:
+                    logger.warning("ALERT: New Tapes needed in slots 13-24")
+                    #sys.exit()#TODO: pause and wait like a good daemon
+                    time.sleep(30)
+                    continue
+                else:
+                    logger.warning("ALERT: New Tapes needed in slots 1-12")
+                    #sys.exit()  #TODO: pause ad wait like a good daemon
+                    time.sleep(30)
+                    continue
+            else:
+                #we have an empty tape lets proceed
+                logger.info(f'loading {newtape_id} from {newslot} to drive {mydrive}')
+                load_tape(mydrive,newslot)
+                logger.info('tape load complete')
+                drivetapeid = newtape_id
+                break
+    if True:
+        #write out the full file list to a txt for offline testing
+        F = open('backup.txt','w')
+        for i in np.arange(len(files)):
+            F.write(f'{drivetapeid},{files[i]}, {obsids[i]}, {start_jds[i]}, {sizes[i]}\n')
+        F.close()
+    logger.info(f'adding {len(files)} files to heratape db')
+    add_files_to_tape(
+        tape_id=drivetapeid,
+        write_date=None,
+        #write_date=Time('2025-01-01'), #missing date implies writing in progress
+        filepath_list=files,
+        obsid_list=obsids,
+        jd_start_list=start_jds,
+        size_list=sizes,
+        testing=TESTING
+    )
+    logger.info('files added, moving on to tape writing')
+    if not TESTING:
+        logger.info("running tar: tar -cjf /dev/st{mydrive} {manyfiles}")
+        tstart = time.time()
+        with subprocess.Popen(f'time tar -cf /dev/nst{mydrive} {files} ', shell=True, stdout=subprocess.PIPE) as proc:                       
+                    lines = proc.stdout.readlines()
+        logger.info(f'finished in {(time.time() - tstart)/60} minutes')
+    else:
+        logger.info(" TESTING MODE: skipping real tar to tape")
+        logger.debug(f' the tar command: time tar -cf /dev/nst{mydrive} {len(files)} not printed')
+    if True:
```

Comment:
why bother with this if statement?

Reply:
the debug log tells you the tar command that would have been run.  But doesn't actually do it.